internal/pkg/service/common/httpserver/httpserver.go

```diff
@@ -48,20 +48,21 @@ func Start(ctx context.Context, d dependencies, cfg Config) error {
 	// Start HTTP server
 	srv := &http.Server{Addr: cfg.ListenAddress, Handler: handler, ReadHeaderTimeout: readHeaderTimeout}
 	proc := d.Process()
-	proc.Add(func(ctx context.Context, shutdown servicectx.ShutdownFn) {
+	proc.Add(func(shutdown servicectx.ShutdownFn) {
 		// Start HTTP server in a separate goroutine.
 		logger.InfofCtx(ctx, "HTTP server listening on %q", cfg.ListenAddress)
-		shutdown(ctx, srv.ListenAndServe())
+		serverErr := srv.ListenAndServe() // ListenAndServe blocks while the server is running
+		shutdown(context.Background(), serverErr)
```

Comment:
:exclamation: :exclamation: :exclamation: shutdown starts here, so the shutdown context is created here :exclamation: :exclamation: :exclamation: